test/shared/src/main/scala/zio/test/experimental/TestAspect.scala

```diff
@@ -0,0 +1,384 @@
+package zio.test.experimental
+
+import scala.annotation.unchecked.uncheckedVariance
+
+import zio._
+import zio.duration._
+import zio.test._
+import zio.test.environment._
+
+trait TestAspect[+EnvIn, -ErrIn, -EnvOut, +ErrOut] { self =>
+  type ModifyEnv[+_ >: EnvIn] >: EnvOut
+  type ModifyErr[+_ <: ErrIn] <: ErrOut
+
+  /**
+   * Applies the aspect to some tests in the spec, chosen by the provided
+   * predicate.
+   */
+  def some[R >: EnvIn, E <: ErrIn](predicate: String => Boolean, spec: ZSpec[R, E]): ZSpec[ModifyEnv[R], ModifyErr[E]]
+
+  /**
+   * An alias for [[all]].
+   */
+  def apply[R >: EnvIn, E <: ErrIn](spec: ZSpec[R, E]): ZSpec[ModifyEnv[R], ModifyErr[E]] =
+    all(spec)
+
+  /**
+   * Applies the aspect to every test in the spec.
+   */
+  def all[R >: EnvIn, E <: ErrIn](spec: ZSpec[R, E]): ZSpec[ModifyEnv[R], ModifyErr[E]] =
+    some(_ => true, spec)
+
+  /**
+   * Returns a new aspect that represents the sequential composition of this
+   * aspect with the specified one.
+   */
+  def >>>[EnvOut1, ErrOut1](that: TestAspect[EnvOut, ErrOut, EnvOut1, ErrOut1]): TestAspect.Aux[
+    EnvIn,
+    ErrIn,
+    EnvOut1,
+    ErrOut1,
+    ({ type Apply[+R >: EnvIn @uncheckedVariance] = that.ModifyEnv[self.ModifyEnv[R]] })#Apply,
```

Comment:
This creates a warning that `EnvIn` and `ErrIn` appear in invariant position on Scala 2.12 but not Scala 2.13.